### PR TITLE
paco-tb-8-translation-history

### PR DESCRIPTION
**Pull request by franciscodr**
This PR adds:
- Repository for saving translations
- Fixes a issue related to an animation

@javipacheco Could you review this code, please? Thanks!

### CODE REVIEW

File: src/main/scala/com/fortysevendeg/translatebubble/modules/repository/impl/RepositoryServicesComponentImpl.scala [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+              message = "The translation history item has been added successfully",
+              translationHistoryEntity = Some(TranslationHistoryEntity(
+                id = Integer.parseInt(uri.getPathSegments.get(1)),
+                data = request.data)))
+
+          } recover {
+            case e: Exception =>
+              AddTranslationHistoryResponse(
+                success = false,
+                message = "Unexpected error when adding a translation history item",
```
Comment (javipacheco) [inline src/main/scala/com/fortysevendeg/translatebubble/modules/repository/impl/RepositoryServicesComponentImpl.scala]: We shouldn't send messages from `Response` if not necessary

File: src/main/scala/com/fortysevendeg/translatebubble/modules/repository/impl/RepositoryServicesComponentImpl.scala [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        tryToFuture {
+          Try {
+            appContextProvider.get.getContentResolver.delete(
+              TranslateBubbleContentProvider.contentUriTranslationHistory,
+              s"${TranslateBubbleSqlHelper.id}=?",
+              Seq(request.entity.id.toString).toArray)
+
+            DeleteTranslationHistoryResponse(
+              success = true,
+              message = "The translation history item has been deleted successfully")
```
Comment (javipacheco) [inline src/main/scala/com/fortysevendeg/translatebubble/modules/repository/impl/RepositoryServicesComponentImpl.scala]: See above

File: src/main/scala/com/fortysevendeg/translatebubble/modules/repository/impl/RepositoryServicesComponentImpl.scala [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+          } recover {
+            case e: Exception =>
+              DeleteTranslationHistoryResponse(
+                success = false,
+                message = "Unexpected error when deleting a translation history item")
```
Comment (javipacheco) [inline src/main/scala/com/fortysevendeg/translatebubble/modules/repository/impl/RepositoryServicesComponentImpl.scala]: See above

File: src/main/scala/com/fortysevendeg/translatebubble/modules/repository/impl/RepositoryServicesComponentImpl.scala [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+import com.fortysevendeg.translatebubble.provider.{TranslateBubbleSqlHelper, TranslateBubbleContentProvider, TranslationHistoryEntity}
+import com.fortysevendeg.translatebubble.service.Service
+import com.fortysevendeg.translatebubble.utils.DBUtils
+import com.fortysevendeg.translatebubble.utils.LanguageTypeTransformer._
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+trait RepositoryServicesComponentImpl extends RepositoryServicesComponent with DBUtils {
```
Comment (javipacheco) [inline src/main/scala/com/fortysevendeg/translatebubble/modules/repository/impl/RepositoryServicesComponentImpl.scala]: Every `with` and  `extends` in a new line

File: src/main/scala/com/fortysevendeg/translatebubble/modules/repository/Messages.scala [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+package com.fortysevendeg.translatebubble.modules.repository
+
+import com.fortysevendeg.translatebubble.provider.{TranslationHistoryEntity, TranslationHistoryEntityData}
+import com.fortysevendeg.translatebubble.utils.LanguageType.LanguageType
+
+case class AddTranslationHistoryRequest(data: TranslationHistoryEntityData)
+
+case class AddTranslationHistoryResponse(success: Boolean, message: String, translationHistoryEntity: Option[TranslationHistoryEntity])
```
Comment (javipacheco) [inline src/main/scala/com/fortysevendeg/translatebubble/modules/repository/Messages.scala]: Every field in a new line

File: src/main/scala/com/fortysevendeg/translatebubble/provider/TranslateBubbleContentProvider.scala [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+package com.fortysevendeg.translatebubble.provider
+
+import android.content.{ContentProvider, ContentUris, ContentValues, UriMatcher}
+import android.database.Cursor
+import android.database.sqlite.{SQLiteDatabase, SQLiteQueryBuilder}
+import android.net.Uri
+import TranslateBubbleContentProvider._
+
+class TranslateBubbleContentProvider extends ContentProvider {
```
Comment (javipacheco) [inline src/main/scala/com/fortysevendeg/translatebubble/provider/TranslateBubbleContentProvider.scala]: `extends` in a new line

File: src/main/scala/com/fortysevendeg/translatebubble/ui/commons/ListLayout.scala [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        (placeholderContent <~ vVisible))
+  }
+
+  def adapter[VH <: RecyclerView.ViewHolder](adapter: RecyclerView.Adapter[VH]) = {
+    runUi(
+      (progressBar <~ vGone) ~
+        (placeholderContent <~ vGone) ~
+        (recyclerView <~ vVisible) ~
+        (recyclerView <~ rvAdapter(adapter))
+    )
```
Comment (javipacheco) [inline src/main/scala/com/fortysevendeg/translatebubble/ui/commons/ListLayout.scala]: Parenthesis above

File: src/main/scala/com/fortysevendeg/translatebubble/ui/history/TranslationHistoryActivity.scala [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+  def reloadList(translationHistoryItems: Seq[TranslationHistoryEntity]) = {
+    translationHistoryItems.length match {
+      case 0 => empty()
+      case _ =>
+        val translationAdapter = new TranslationHistoryAdapter(translationHistoryItems, new RecyclerClickListener {
+          override def onClick(translationHistoryItem: TranslationHistoryEntity): Unit = {}
+        })
+        adapter(translationAdapter)
+    }
+  }
```
Comment (javipacheco) [inline src/main/scala/com/fortysevendeg/translatebubble/ui/history/TranslationHistoryActivity.scala]: Remove braces

File: src/main/scala/com/fortysevendeg/translatebubble/modules/repository/impl/RepositoryServicesComponentImpl.scala (hunk 1 above)
Comment (franciscodr) [inline src/main/scala/com/fortysevendeg/translatebubble/modules/repository/impl/RepositoryServicesComponentImpl.scala]: Agree with you, I'll remove it

**Comment by javipacheco** [comment]
LGTM!

**Pull request merged**